.github/workflows/run-kata-coco-tests.yaml

```diff
@@ -38,6 +38,7 @@ jobs:
       KATA_HYPERVISOR: ${{ matrix.vmm }}
       KUBERNETES: "k3s"
       USING_NFD: "true"
+      GUEST_PULL: "yes"
```

Comment:
I don't think this `GUEST_PULL` env var will be enough.

Ideally we'd have to env vars:
* `SNAPSHOTTER`, to follow what we're doing for containerd tests.
* And one for `PULL_TYPE`, which would be either `guest` or `host`.

Reply:
On this, snapshotter and pull-type could be part of the matrix.  For now, only `nydus` will be there, but this could be more easily expandable to support MSFT's approach in the future.

Reply:
Fixed.